node/node.go

```diff
@@ -181,18 +202,22 @@ func (nn *nexNode) initializeSupervisionTree() (goakt.ActorSystem, error) {
 	if err != nil {
 		return nil, err
 	}
-	inats := actors.CreateInternalNatsServer(*nn.options)
+	nn.runningActors[actors.AgentSupervisorActorName] = agentSuper
```

Comment:
we definitely shouldn't have to maintain a map of actors. We can look up any actor by its name, and we can obtain its list of children with the `Children` function.

Reply:
learning the hard way!